libs/libc/pthread/pthread_keydelete.c

```diff
@@ -52,8 +57,33 @@
 
 int pthread_key_delete(pthread_key_t key)
 {
-  /* Free the TLS index */
+  FAR struct task_info_s *info = task_get_info();
+  tls_ndxset_t mask;
+  int ret = EINVAL;
+
+  DEBUGASSERT(info != NULL);
+  DEBUGASSERT(key >= 0 && key < CONFIG_TLS_NELEM);
+  if (key >= 0 && key < CONFIG_TLS_NELEM)
+    {
+      /* This is done while holding a semaphore here to avoid concurrent
+       * modification of the group TLS index set.
+       */
 
-  int ret = tls_free((int)key);
-  return ret < 0 ? -ret : 0;
+      mask = (tls_ndxset_t)1 << key;
+      ret = _SEM_WAIT(&info->ta_sem);
```

Comment:
I think we need to wait for https://github.com/apache/incubator-nuttx/pull/6320 to be merged first as according to POSIX: `The pthread_key_delete() function shall not return an error code of [EINTR].`

Reply:
@pkarashchenko could you merge https://github.com/apache/incubator-nuttx-apps/pull/1173 first to unblock the ci?

Reply:
Done